### PR TITLE
efm-langserver: init at 0.0.31

### PR DESCRIPTION
**Pull request by Philipp-M**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 121384` at cca4c6cb run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built successfully:</summary>
  <ul>
    <li>efm-langserver</li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/no-uri-literals.md">no-uri-literals</a>

URI literals are deprecated.
Near pkgs/development/tools/efm-langserver/default.nix:20:16:
```
   |
20 |     homepage = https://github.com/mattn/efm-langserver;
   |                ^
```
</li>
  </ul>
</details>

File: pkgs/development/tools/efm-langserver/default.nix [hunk 1]
```diff
@@ -0,0 +1,24 @@
+{ lib, buildGoModule, fetchFromGitHub }:
+
+buildGoModule rec {
+  name = "efm-langserver";
+  version = "0.0.29";
```
Comment (jonringer) [inline pkgs/development/tools/efm-langserver/default.nix:5]:
```suggestion
  version = "0.0.29";
  
```

File: pkgs/development/tools/efm-langserver/default.nix [hunk 2]
```diff
@@ -0,0 +1,24 @@
+{ lib, buildGoModule, fetchFromGitHub }:
+
+buildGoModule rec {
+  name = "efm-langserver";
```
Comment (jonringer) [inline pkgs/development/tools/efm-langserver/default.nix:4]:
```suggestion
  pname = "efm-langserver";
```

File: pkgs/development/tools/efm-langserver/default.nix [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+  version = "0.0.29";
+  src = fetchFromGitHub {
+    owner = "mattn";
+    repo = "efm-langserver";
+    rev = "v${version}";
+    sha256 = "sha256-/G3GF4tz0MzR74aLKViKm03jT8e1MOzAjSpq9Aof5n8=";
+  };
+
+  modSha256 = vendorSha256;
+  vendorSha256 = "sha256-tca+1SRrFyvU8ttHmfMFiGXd1A8rQSEWm1Mc2qp0EfI=";
```
Comment (jonringer) [inline pkgs/development/tools/efm-langserver/default.nix:14]: only vendorSha is used for buildGoModule https://nixos.org/manual/nixpkgs/stable/#ssec-go-modules
```suggestion
  vendorSha256 = "sha256-tca+1SRrFyvU8ttHmfMFiGXd1A8rQSEWm1Mc2qp0EfI=";
```

**Comment by bnjmnt4n** [comment]
Is anything blocking this? Would be great to see this merged!

File: pkgs/development/tools/efm-langserver/default.nix [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+
+  vendorSha256 = "sha256-tca+1SRrFyvU8ttHmfMFiGXd1A8rQSEWm1Mc2qp0EfI=";
+  subPackages = [ "." ];
+
+  meta = with lib; {
+    description = "General purpose Language Server";
+    maintainers = with maintainers; [ Philipp-M ];
+    homepage = "https://github.com/mattn/efm-langserver";
+    license = licenses.mit;
+    platforms = platforms.linux ++ platforms.darwin;
```
Comment (SuperSandro2000) [inline pkgs/development/tools/efm-langserver/default.nix:22]:
```suggestion
```

**Pull request merged**